src/shared/ui/button/Button.tsx

```diff
@@ -0,0 +1,27 @@
+import styles from './button.module.css';
+import React from "react";
+
+type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  size?: '' | 'semi' | 'large';
+  variant?: string;
+  onClick?: () => void;
```

Comment:
remove onClick please
this is already exists in ButtonHTMLAttributes

Reply:
Without declaration TS very angry on this "onClick" function